src/librustdoc/html/render/shared_context.rs

```diff
@@ -0,0 +1,72 @@
+use std::cell::RefCell;
+use std::path::{Path, PathBuf};
+
+use rustc_data_structures::fx::{FxHashMap, FxHashSet};
+use rustc_middle::ty::TyCtxt;
+use rustc_span::edition::Edition;
+
+use super::StylePath;
+use crate::clean;
+use crate::docfs::{DocFS, PathError};
+use crate::error::Error;
+use crate::html::layout;
+use crate::html::markdown::{self, ErrorCodes};
+
+crate struct SharedContext<'tcx> {
```

Comment:
It seems a little silly to put this in its own file when it's only 72 lines.

Reply:
So `SharedContext` should be re-merged with the `mod.rs`.